src/fudge/log.clj

```diff
@@ -118,40 +130,62 @@
 
 ;; # Main logging functions
 
-(defn log
-  "Log a record with the config, level, and data provided"
-  [c level data]
-  (let [record (invoke c :prepare-fn data level)]
+(defn log!
```

Comment:
Ah, if just to distinguish from the log macro, I've seen people use `log-fn` for the function implementation (though that's uncomfortable here, as we're using `-fn` for the callbacks already.)  Perhaps separating macros into another namespace as I mentioned elsewhere is the thing.